backend/src/test/scala/com/softwaremill/bootzooka/test/TestHelpersWithDb.scala

```diff
@@ -4,20 +4,22 @@ import com.softwaremill.bootzooka.common.sql.SqlDatabase
 import com.softwaremill.bootzooka.email.application.{DummyEmailService, EmailTemplatingEngine}
 import com.softwaremill.bootzooka.user.application.{UserDao, UserService}
 import com.softwaremill.bootzooka.user.domain.User
+import org.scalatest.AsyncFlatSpec
 import org.scalatest.concurrent.ScalaFutures
-
 import scala.concurrent.ExecutionContext
 
 trait TestHelpersWithDb extends TestHelpers with ScalaFutures {
 
+  this: AsyncFlatSpec =>
+
   lazy val emailService = new DummyEmailService()
   lazy val emailTemplatingEngine = new EmailTemplatingEngine
-  lazy val userDao = new UserDao(sqlDatabase)
-  lazy val userService = new UserService(userDao, emailService, emailTemplatingEngine)
+  lazy val initEc = ExecutionContext.global
 
-  def sqlDatabase: SqlDatabase
+  lazy val userDao = new UserDao(sqlDatabase)(initEc)
```

Comment:
You can use it in your tests without problems. I think they just didn't predict that someone may need EC outside actual test body. Or I'm not getting something?
From docs:
_When ScalaTest's serial execution context is called upon to execute a task, that task is recorded in a queue for later execution. For example, one task that will be placed in this queue is the task that transforms the Future[Assertion] returned by an asynchronous test body to the FutureOutcome returned from the test function. Other tasks that will be queued are any transformations of, or callbacks registered on, Futures that occur in your test body, including any assertions you map onto Futures. Once the test body returns, the thread that executed the test body will execute the tasks in that queue one after another, in the order they were enqueued._
http://doc.scalatest.org/3.0.0/index.html#org.scalatest.AsyncFlatSpec